app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt

```diff
@@ -234,9 +234,15 @@ class NgVpnNetworkStack @Inject constructor(
 
             tunnelThread = Thread {
                 logcat { "Running tunnel in context $jniContext" }
-                vpnNetwork.run(jniContext, tunfd)
-                logcat(LogPriority.WARN) { "Tunnel exited" }
-                tunnelThread = null
+                try {
+                    vpnNetwork.run(jniContext, tunfd)
+                } catch (t: Throwable) {
+                    logcat(LogPriority.ERROR) { "Tunnel thread crashed: ${t.asLog()}" }
+                    deviceShieldPixels.reportTunnelThreadAbnormalCrash()
+                } finally {
+                    tunnelThread = null
```

Comment:
Could this cause a `nullpointer` in line 246 `it.start()`?

Reply:
The assignment to `tunnelThread` has already happened before the thread ever sets `tunnelThread = null` inside its body. Example

```
val thread = Thread {
    // Runs later
}.also { it.start() } // Starts immediately

// thread is still a valid reference here
```

Does it make sense?

Reply:
Ah makes sense 👍